{
  "totalTests": 96,
  "passingTests": 63,
  "failingTests": 33,
  "totalFiles": 46,
  "passingFiles": 15,
  "failingFiles": 31,
  "failedSanity": 1,
  "passedSanity": 46,
  "failedSanityTests": [
    {
      "title": "sanity check: reentrancy/reentrancy_bonus.sol",
      "file": "reentrancy/reentrancy_bonus_test.js",
      "contractFile": "reentrancy/reentrancy_bonus.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x01e21d7b8c39dc4c764c19b308bd8b14b1ba139e)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.getFirstWithdrawalBonus (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/reentrancy/reentrancy_bonus_test.js:40:16)"
    }
  ],
  "passedResults": [
    {
      "title": "exploit access control vulnerability",
      "file": "access_control/FibonacciBalance_test.js",
      "contractFile": "access_control/FibonacciBalance.sol",
      "state": "passed"
    },
    {
      "title": "exploit access control vulnerability",
      "file": "access_control/wallet_02_refund_nosub_test.js",
      "contractFile": "access_control/wallet_02_refund_nosub.sol",
      "state": "passed"
    },
    {
      "title": "exploit access control vulnerability",
      "file": "access_control/wallet_03_wrong_constructor_test.js",
      "contractFile": "access_control/wallet_03_wrong_constructor.sol",
      "state": "passed"
    },
    {
      "title": "exploit bad randomness vulnerability",
      "file": "bad_randomness/old_blockhash_test.js",
      "contractFile": "bad_randomness/old_blockhash.sol",
      "state": "passed"
    },
    {
      "title": "exploit denial of service vulnerability",
      "file": "denial_of_service/auction_test.js",
      "contractFile": "denial_of_service/auction.sol",
      "state": "passed"
    },
    {
      "title": "front running vulnerability in setReward() function",
      "file": "front_running/eth_tx_order_dependence_minimal_test.js",
      "contractFile": "front_running/eth_tx_order_dependence_minimal.sol",
      "state": "passed"
    },
    {
      "title": "front running vulnerability in claimReward() function",
      "file": "front_running/eth_tx_order_dependence_minimal_test.js",
      "contractFile": "front_running/eth_tx_order_dependence_minimal.sol",
      "state": "passed"
    },
    {
      "title": "front running vulnerability",
      "file": "front_running/odds_and_evens_test.js",
      "contractFile": "front_running/odds_and_evens.sol",
      "state": "passed"
    },
    {
      "title": "exploit uninitialized storage vulnerability",
      "file": "other/crypto_roulette_test.js",
      "contractFile": "other/crypto_roulette.sol",
      "state": "passed"
    },
    {
      "title": "exploit time manipulation vulnerability",
      "file": "time_manipulation/ether_lotto_test.js",
      "contractFile": "time_manipulation/ether_lotto.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability",
      "file": "unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e_test.js",
      "contractFile": "unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability in WithdrawToken()",
      "file": "unchecked_low_level_calls/0x627fa62ccbb1c1b04ffaecd72a53e37fc0e17839_test.js",
      "contractFile": "unchecked_low_level_calls/0x627fa62ccbb1c1b04ffaecd72a53e37fc0e17839.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability",
      "file": "unchecked_low_level_calls/0x7d09edb07d23acb532a82be3da5c17d9d85806b4_test.js",
      "contractFile": "unchecked_low_level_calls/0x7d09edb07d23acb532a82be3da5c17d9d85806b4.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability in line 192",
      "file": "unchecked_low_level_calls/0x89c1b3807d4c67df034fffb62f3509561218d30b_test.js",
      "contractFile": "unchecked_low_level_calls/0x89c1b3807d4c67df034fffb62f3509561218d30b.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability in line 180",
      "file": "unchecked_low_level_calls/0x89c1b3807d4c67df034fffb62f3509561218d30b_test.js",
      "contractFile": "unchecked_low_level_calls/0x89c1b3807d4c67df034fffb62f3509561218d30b.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability",
      "file": "unchecked_low_level_calls/0xb7c5c5aa4d42967efe906e1b66cb8df9cebf04f7_test.js",
      "contractFile": "unchecked_low_level_calls/0xb7c5c5aa4d42967efe906e1b66cb8df9cebf04f7.sol",
      "state": "passed"
    },
    {
      "title": "exploit unchecked low level call vulnerability",
      "file": "unchecked_low_level_calls/0xe894d54dca59cb53fe9cbc5155093605c7068220_test.js",
      "contractFile": "unchecked_low_level_calls/0xe894d54dca59cb53fe9cbc5155093605c7068220.sol",
      "state": "passed"
    }
  ],
  "failedResults": [
    {
      "title": "exploit access control vulnerability",
      "file": "access_control/mycontract_test.js",
      "contractFile": "access_control/mycontract.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x0165878a594ca255338adfa4d48449f69242eb8f)\n    at <UnrecognizedContract>.<unknown> (0xa513e6e4b8f2a923d98304ec87f64353c4d5c853)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at Context.<anonymous> (test/access_control/mycontract_test.js:63:5)"
    },
    {
      "title": "exploit access control vulnerability",
      "file": "access_control/phishable_test.js",
      "contractFile": "access_control/phishable.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x8a791620dd6260079bf849dc5567adc3f2fdc318)\n    at <UnrecognizedContract>.<unknown> (0x610178da211fef7d417bc0e6fed39f05609ad788)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at Context.<anonymous> (test/access_control/phishable_test.js:72:5)"
    },
    {
      "title": "exploit access control vulnerability",
      "file": "access_control/wallet_04_confused_sign_test.js",
      "contractFile": "access_control/wallet_04_confused_sign.sol",
      "state": "failed",
      "error": "VM Exception while processing transaction: invalid opcode",
      "stack": "Error: VM Exception while processing transaction: invalid opcode\n    at <UnrecognizedContract>.<unknown> (0x322813fd9a801c5507c9de605d63cea4f2ce6c44)\n    at <UnrecognizedContract>.<unknown> (0xa85233c63b9ee964add6f2cffe00fd84eb32338f)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/access_control/wallet_04_confused_sign_test.js:71:5)"
    },
    {
      "title": "exploit overflow vulnerability",
      "file": "arithmetic/timelock_test.js",
      "contractFile": "arithmetic/timelock.sol",
      "state": "failed",
      "error": "VM Exception while processing transaction: invalid opcode",
      "stack": "Error: VM Exception while processing transaction: invalid opcode\n    at <UnrecognizedContract>.<unknown> (0x09635f643e140090a9a8dcd712ed6285858cebef)\n    at <UnrecognizedContract>.<unknown> (0xc5a5c42992decbae36851359345fe25997f5c42d)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/arithmetic/timelock_test.js:57:5)"
    },
    {
      "title": "exploit buy overflow vulnerability line 23",
      "file": "arithmetic/tokensalechallenge_test.js",
      "contractFile": "arithmetic/tokensalechallenge.sol",
      "state": "failed",
      "error": "VM Exception while processing transaction: invalid opcode",
      "stack": "Error: VM Exception while processing transaction: invalid opcode\n    at <UnrecognizedContract>.<unknown> (0x84ea74d481ee0a5332c457a4d796187f6ba67feb)\n    at <UnrecognizedContract>.<unknown> (0xc3e53f4d16ae77db1c982e75a937b9f60fe63690)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack_buy (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/arithmetic/tokensalechallenge_test.js:49:5)"
    },
    {
      "title": "exploit the catch the ether vulnerability",
      "file": "arithmetic/tokensalechallenge_test.js",
      "contractFile": "arithmetic/tokensalechallenge.sol",
      "state": "failed",
      "error": "VM Exception while processing transaction: invalid opcode",
      "stack": "Error: VM Exception while processing transaction: invalid opcode\n    at <UnrecognizedContract>.<unknown> (0x84ea74d481ee0a5332c457a4d796187f6ba67feb)\n    at <UnrecognizedContract>.<unknown> (0xc3e53f4d16ae77db1c982e75a937b9f60fe63690)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack_complete (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/arithmetic/tokensalechallenge_test.js:60:5)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x01f8c4e3fa3edeb29e514cba738d87ce8c091d3f_test.js",
      "contractFile": "reentrancy/0x01f8c4e3fa3edeb29e514cba738d87ce8c091d3f.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x01f8c4e3fa3edeb29e514cba738d87ce8c091d3f_test.js:90:39)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x23a91059fdc9579a9fbd0edc5f2ea0bfdb70deb4_test.js",
      "contractFile": "reentrancy/0x23a91059fdc9579a9fbd0edc5f2ea0bfdb70deb4.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x23a91059fdc9579a9fbd0edc5f2ea0bfdb70deb4_test.js:89:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x4320e6f8c05b27ab4707cd1f6d5ce6f3e4b3a5a1_test.js",
      "contractFile": "reentrancy/0x4320e6f8c05b27ab4707cd1f6d5ce6f3e4b3a5a1.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x4320e6f8c05b27ab4707cd1f6d5ce6f3e4b3a5a1_test.js:91:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x4e73b32ed6c35f570686b89848e5f39f20ecc106_test.js",
      "contractFile": "reentrancy/0x4e73b32ed6c35f570686b89848e5f39f20ecc106.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x4e73b32ed6c35f570686b89848e5f39f20ecc106_test.js:91:39)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x561eac93c92360949ab1f1403323e6db345cbf31_test.js",
      "contractFile": "reentrancy/0x561eac93c92360949ab1f1403323e6db345cbf31.sol",
      "state": "failed",
      "error": "expected 8000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 8000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x561eac93c92360949ab1f1403323e6db345cbf31_test.js:87:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x7541b76cb60f4c60af330c208b0623b7f54bf615_test.js",
      "contractFile": "reentrancy/0x7541b76cb60f4c60af330c208b0623b7f54bf615.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x7541b76cb60f4c60af330c208b0623b7f54bf615_test.js:92:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x7a8721a9d64c74da899424c1b52acbf58ddc9782_test.js",
      "contractFile": "reentrancy/0x7a8721a9d64c74da899424c1b52acbf58ddc9782.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x7a8721a9d64c74da899424c1b52acbf58ddc9782_test.js:91:39)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x7b368c4e805c3870b6c49a3f1f49f69af8662cf3_test.js",
      "contractFile": "reentrancy/0x7b368c4e805c3870b6c49a3f1f49f69af8662cf3.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x7b368c4e805c3870b6c49a3f1f49f69af8662cf3_test.js:90:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x8c7777c45481dba411450c228cb692ac3d550344_test.js",
      "contractFile": "reentrancy/0x8c7777c45481dba411450c228cb692ac3d550344.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x8c7777c45481dba411450c228cb692ac3d550344_test.js:92:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x93c32845fae42c83a70e5f06214c8433665c2ab5_test.js",
      "contractFile": "reentrancy/0x93c32845fae42c83a70e5f06214c8433665c2ab5.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x93c32845fae42c83a70e5f06214c8433665c2ab5_test.js:93:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x941d225236464a25eb18076df7da6a91d0f95e9e_test.js",
      "contractFile": "reentrancy/0x941d225236464a25eb18076df7da6a91d0f95e9e.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x941d225236464a25eb18076df7da6a91d0f95e9e_test.js:91:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0x96edbe868531bd23a6c05e9d0c424ea64fb1b78b_test.js",
      "contractFile": "reentrancy/0x96edbe868531bd23a6c05e9d0c424ea64fb1b78b.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0x96edbe868531bd23a6c05e9d0c424ea64fb1b78b_test.js:87:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xaae1f51cf3339f18b6d3f3bdc75a5facd744b0b8_test.js",
      "contractFile": "reentrancy/0xaae1f51cf3339f18b6d3f3bdc75a5facd744b0b8.sol",
      "state": "failed",
      "error": "expected 9000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 9000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xaae1f51cf3339f18b6d3f3bdc75a5facd744b0b8_test.js:91:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xb5e1b1ee15c6fa0e48fce100125569d430f1bd12_test.js",
      "contractFile": "reentrancy/0xb5e1b1ee15c6fa0e48fce100125569d430f1bd12.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xb5e1b1ee15c6fa0e48fce100125569d430f1bd12_test.js:90:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xb93430ce38ac4a6bb47fb1fc085ea669353fd89e_test.js",
      "contractFile": "reentrancy/0xb93430ce38ac4a6bb47fb1fc085ea669353fd89e.sol",
      "state": "failed",
      "error": "expected 7000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 7000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xb93430ce38ac4a6bb47fb1fc085ea669353fd89e_test.js:89:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xbaf51e761510c1a11bf48dd87c0307ac8a8c8a4f_test.js",
      "contractFile": "reentrancy/0xbaf51e761510c1a11bf48dd87c0307ac8a8c8a4f.sol",
      "state": "failed",
      "error": "expected 9000000000000000000 to be below 4000000000000000000.",
      "stack": "AssertionError: expected 9000000000000000000 to be below 4000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xbaf51e761510c1a11bf48dd87c0307ac8a8c8a4f_test.js:89:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xbe4041d55db380c5ae9d4a9b9703f1ed4e7e3888_test.js",
      "contractFile": "reentrancy/0xbe4041d55db380c5ae9d4a9b9703f1ed4e7e3888.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xbe4041d55db380c5ae9d4a9b9703f1ed4e7e3888_test.js:91:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xcead721ef5b11f1a7b530171aab69b16c5e66b6e_test.js",
      "contractFile": "reentrancy/0xcead721ef5b11f1a7b530171aab69b16c5e66b6e.sol",
      "state": "failed",
      "error": "expected 10000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 10000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xcead721ef5b11f1a7b530171aab69b16c5e66b6e_test.js:90:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/0xf015c35649c82f5467c9c74b7f28ee67665aad68_test.js",
      "contractFile": "reentrancy/0xf015c35649c82f5467c9c74b7f28ee67665aad68.sol",
      "state": "failed",
      "error": "expected 8000000000000000000 to be below 3000000000000000000.",
      "stack": "AssertionError: expected 8000000000000000000 to be below 3000000000000000000.\n    at Context.<anonymous> (test/reentrancy/0xf015c35649c82f5467c9c74b7f28ee67665aad68_test.js:90:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/etherstore_test.js",
      "contractFile": "reentrancy/etherstore.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x8d81a3dcd17030cd5f23ac7370e4efb10d2b3ca4)\n    at <UnrecognizedContract>.<unknown> (0xcc4c41415fc68b2fbf70102742a83cde435e0ca7)\n    at <UnrecognizedContract>.<unknown> (0x8d81a3dcd17030cd5f23ac7370e4efb10d2b3ca4)\n    at <UnrecognizedContract>.<unknown> (0xcc4c41415fc68b2fbf70102742a83cde435e0ca7)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/reentrancy/etherstore_test.js:73:5)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/modifier_reentrancy_test.js",
      "contractFile": "reentrancy/modifier_reentrancy.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x871acbeabbaf8bed65c22ba7132becfabf8c27b5)\n    at <UnrecognizedContract>.<unknown> (0xc1e0a9db9ea830c52603798481045688c8ae99c2)\n    at <UnrecognizedContract>.<unknown> (0x871acbeabbaf8bed65c22ba7132becfabf8c27b5)\n    at <UnrecognizedContract>.<unknown> (0xc1e0a9db9ea830c52603798481045688c8ae99c2)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.initiateAttack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/reentrancy/modifier_reentrancy_test.js:53:5)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/reentrance_test.js",
      "contractFile": "reentrancy/reentrance.sol",
      "state": "failed",
      "error": "expected 9000000000000000000 to be below 5000000000000000000.",
      "stack": "AssertionError: expected 9000000000000000000 to be below 5000000000000000000.\n    at Context.<anonymous> (test/reentrancy/reentrance_test.js:80:33)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/reentrancy_bonus_test.js",
      "contractFile": "reentrancy/reentrancy_bonus.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x1343248cbd4e291c6979e70a138f4c774e902561)\n    at <UnrecognizedContract>.<unknown> (0x22a9b82a6c3d2bfb68f324b2e8367f346dd6f32a)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/reentrancy/reentrancy_bonus_test.js:64:5)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/reentrancy_dao_test.js",
      "contractFile": "reentrancy/reentrancy_dao.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x5e6cb7e728e1c320855587e1d9c6f7972ebdd6d5)\n    at <UnrecognizedContract>.<unknown> (0x79e8ab29ff79805025c9462a2f2f12e9a496f81d)\n    at <UnrecognizedContract>.<unknown> (0x5e6cb7e728e1c320855587e1d9c6f7972ebdd6d5)\n    at <UnrecognizedContract>.<unknown> (0x79e8ab29ff79805025c9462a2f2f12e9a496f81d)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/reentrancy/reentrancy_dao_test.js:70:5)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/reentrancy_simple_test.js",
      "contractFile": "reentrancy/reentrancy_simple.sol",
      "state": "failed",
      "error": "Transaction reverted without a reason string",
      "stack": "Error: Transaction reverted without a reason string\n    at <UnrecognizedContract>.<unknown> (0x512f7469bcc83089497506b5df64c6e246b39925)\n    at <UnrecognizedContract>.<unknown> (0x9fd16ea9e31233279975d99d5e8fc91dd214c7da)\n    at <UnrecognizedContract>.<unknown> (0x512f7469bcc83089497506b5df64c6e246b39925)\n    at <UnrecognizedContract>.<unknown> (0x9fd16ea9e31233279975d99d5e8fc91dd214c7da)\n    at EdrProviderWrapper.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:446:41)\n    at HardhatEthersSigner.sendTransaction (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:125:18)\n    at send (node_modules/ethers/src.ts/contract/contract.ts:313:20)\n    at Proxy.attack (node_modules/ethers/src.ts/contract/contract.ts:352:16)\n    at Context.<anonymous> (test/reentrancy/reentrancy_simple_test.js:72:5)"
    },
    {
      "title": "should successfully drain funds through reentrancy attack",
      "file": "reentrancy/simple_dao_test.js",
      "contractFile": "reentrancy/simple_dao.sol",
      "state": "failed",
      "error": "expected 0 to be above 1000000000000000000.",
      "stack": "AssertionError: expected 0 to be above 1000000000000000000.\n    at Context.<anonymous> (test/reentrancy/simple_dao_test.js:90:44)"
    }
  ]
}